src/avro/reader.rs

```diff
@@ -119,24 +119,20 @@ impl<R: AsyncRead + Unpin + Send> Block<R> {
                 }
             }
 
-            self.codec = meta
-                .get("avro.codec")
-                .ok_or_else(|| DecodeError::new("unable to parse codec: 'avro.codec' missing"))
-                .and_then(|codec| {
-                    if let Value::Bytes(ref bytes) = *codec {
-                        from_utf8(bytes.as_ref())
-                            .map_err(|e| DecodeError::new(format!("unable to decode codec: {}", e)))
-                    } else {
-                        Err(DecodeError::new(format!(
-                            "unable to parse codec: expected bytes, got: {:?}",
-                            codec
-                        )))
-                    }
-                })
-                .and_then(|codec| {
-                    Codec::from_str(codec)
-                        .map_err(|_| DecodeError::new(format!("unrecognized codec '{}'", codec)))
-                })?;
+            self.codec = match meta.get("avro.codec") {
```

Comment:
I think it is clearer to express the defaulting behavior using `unwrap_or` or `unwrap_or_else`, rather than explicitly matching on the option and handling `None`. TIOLI

Reply:
You mean like so?

```rust
meta
                .get("avro.codec")
                .map(|val| match val {
                    Value::Bytes(ref bytes) => from_utf8(bytes.as_ref())
                        .map_err(|e| DecodeError::new(format!("unable to decode codec: {}", e)))
                        .and_then(|codec| {
                            Codec::from_str(codec).map_err(|_| {
                                DecodeError::new(format!("unrecognized codec '{}'", codec))
                            })
                        }),
                    codec => Err(DecodeError::new(format!(
                        "unable to parse codec: expected bytes, got: {:?}",
                        codec
                    ))),
                })
                .unwrap_or(Ok(Codec::Null))?;
```

Reply:
Yes, I think that's the best way. I sort of want to lie and claim that's what I meant. What I actually meant is this:

```
            self.codec = match meta
                .get("avro.codec")
                .unwrap_or_else(|| &Value::Bytes("null".to_string().into_bytes()))
            {
                Value::Bytes(ref bytes) => from_utf8(bytes.as_ref())
                    .map_err(|e| DecodeError::new(format!("unable to decode codec: {}", e)))
                    .and_then(|codec| {
                        Codec::from_str(codec).map_err(|_| {
                            DecodeError::new(format!("unrecognized codec '{}'", codec))
                        })
                    }),
                codec => Err(DecodeError::new(format!(
                    "unable to parse codec: expected bytes, got: {:?}",
                    codec
                ))),
            }?;
```

But I think what you just suggested is better than your original code, and also better than mine.

Reply:
Anyway, up to you!